src/main/java/nz/ac/auckland/scriptella/driver/http/HTTPConnection.java

```diff
@@ -1,288 +0,0 @@
-
-        try {
-            br = new BufferedReader(resource.open());
-            logger.debug("Resource opened for reading.");
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-
-        String line;
-        List<NameValuePair> nameValuePairList = new ArrayList<>(1);
-
-        try {
-            while ((line = br.readLine()) != null) {
-                String[] components = line.trim().split("=");
-                if (components.length > 1) {
-                    // Remove "$" and double-quotes used to make driver work as user would expect based on csv driver and others.
-
-                    String key = components[1].replace("$", "").replace("?", "").replace("\"", "");
-                    logger.debug("Stripped \'$\' and \'?\' from variables.");
-                    if (parameters.containsKey(key)) {
-                        nameValuePairList.add(new BasicNameValuePair(components[0], (String) parameters.getParameter(key)));
-
-                        logger.debug("Variable found, added to List.");
-                    } else {
-                        nameValuePairList.add(new BasicNameValuePair(components[0], components[1]));
-                        logger.debug("Line contains no variables, adding to list as literal.");
-                    }
-                }
-            }
-        } catch (IOException e) {
-            logger.error("Error occurred while generating request parameters: ", e);
-            throw new RuntimeException(e);
-        }
-
-        return nameValuePairList;
-    }
-
-
-    private String parseJSON(Resource resource, ParametersCallbackMap parameters) {
-
-        BufferedReader br = null;
-        try {
-            br = new BufferedReader(resource.open());
-            logger.debug("Resource opened for reading.");
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-
-        // Using an array list as "\n" in strings tends to break things.
-        List<String> jsonString = new ArrayList<>();
-        String line;
-
-        try {
-            while ((line = br.readLine()) != null) {
-                jsonString.add(line);
-            }
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-
-        StringBuilder parsedJSON = new StringBuilder("");
-
-        for (String jsonLine : jsonString) {
-            // Check if line has a comma as it is stripped out if value is replaced.
-            String comma = jsonLine.contains(",") ? "," : "";
-
-            String[] split = jsonLine.trim().split(":");
-            if (split.length > 1) {
-                // Check if value is meant to be a variable and not literal value with "$" in it.
-                if (split[1].contains("$") && !split[1].contains("\"")) {
-                    String key = split[1].trim().replace("$", "").replace(",", "");
-                    split[1] = ("\"" + ((String) parameters.getParameter(key)) + "\"" + comma);
```

Comment:
and in comparison to the time taken to do the http request and parse it, minimal.